src/pyodide/python-entrypoint-helper.ts

```diff
@@ -24,16 +23,16 @@ import { default as Limiter } from "pyodide-internal:limiter";
 import { entropyBeforeRequest } from "pyodide-internal:topLevelEntropy/lib";
 import { loadPackages } from "pyodide-internal:loadPackage";
 
-function pyimportMainModule(pyodide) {
+function pyimportMainModule(pyodide: Pyodide): PyModule {
   if (!MAIN_MODULE_NAME.endsWith(".py")) {
     throw new Error("Main module needs to end with a .py file extension");
   }
   const mainModuleName = MAIN_MODULE_NAME.slice(0, -3);
   return pyodide.pyimport(mainModuleName);
 }
 
-let pyodidePromise;
-function getPyodide() {
+let pyodidePromise: Promise<Pyodide> | undefined;
```

Comment:
`null` may be better here:

```suggestion
let pyodidePromise: Promise<Pyodide> | null;
```

Reply:
I think `undefined` is more accurate. If you say `let x;` it is the same as `let x = undefined;`.